libbeat/dashboards/modify_json.go

```diff
@@ -175,54 +174,99 @@ func ReplaceIndexInDashboardObject(index string, content []byte) []byte {
 		return content
 	}
 
-	var result []byte
-	r := bufio.NewReader(bytes.NewReader(content))
-	for {
-		line, err := r.ReadBytes('\n')
-		if err != nil {
-			if err == io.EOF {
-				return append(result, replaceInNDJSON(logger, index, line)...)
-			}
-			logger.Error("Error reading bytes from raw dashboard object: %+v", err)
-			return content
-		}
-		result = append(result, replaceInNDJSON(logger, index, line)...)
+	if len(bytes.TrimSpace(content)) == 0 {
+		return content
+	}
+
+	objectMap := make(map[string]interface{}, 0)
+	err := json.Unmarshal(content, &objectMap)
+	if err != nil {
+		logger.Errorf("Failed to convert bytes to map[string]interface: %+v", err)
+		return content
+	}
+
+	attributes, ok := objectMap["attributes"].(map[string]interface{})
+	if !ok {
+		logger.Errorf("Object does not have attributes key")
+		return content
+	}
+
+	if kibanaSavedObject, ok := attributes["kibanaSavedObjectMeta"].(map[string]interface{}); ok {
+		ba := ReplaceIndexInSavedObject(logger, index, kibanaSavedObject)
+		attributes["kibanaSavedObjectMeta"] = ba
+	}
+
+	if visState, ok := attributes["visState"].(string); ok {
+		nya := ReplaceIndexInVisState(logger, index, visState)
+		attributes["visState"] = nya
```

Comment:
Do `ba` and `nya` mean something? :slightly_smiling_face:

Reply:
Independently no, but together "banya" means old, mean woman :D :D :D